rpc/jsoncodec/doc.go

```diff
@@ -1,5 +1,12 @@
-// Copyright 2022 Canonical Ltd.
+// Copyright 2025 Canonical Ltd.
 // Licensed under the AGPLv3, see LICENCE file for details.
 
-// Package jsoncodec provides a JSON codec for the rpc package.
+// Package jsoncodec defines the methods for reading and writing RPC messages (in JSON format)
+// and the connection (gorilla websocket) which passes them. It includes:
+//
+// - Methods for sending and receiving JSON messages.
+// - Definitions for message formats (inMsgV0, inMsgV1, outMsgV0, outMsgV1) across protocol versions.
+// - A mutex to ensure thread-safe access and prevent race conditions.
```

Comment:
todo: delete me

this is a basic definition of why you'd use a mutex and not of particular interest here. the internal comment which says why it's needed is more interesting.